syntax = "proto3";
package cdm;
option java_package = "com.kitware.physiology.cdm";
option optimize_for = SPEED;

message eAnesthesiaMachine
{

  /** @brief An enumeration for the source of oxygen for the machine. */
  enum Event
  {
    OxygenBottleOneExhausted  = 0;
    OxygenBottleTwoExhausted  = 1;
    ReliefValveActive         = 2;
  }

  /** @brief An enumeration for the primary gas used by the machine. */
  enum OxygenSource
  {
    NullSource   = 0;/**<< Signals not provided, or no change */
    NoSource     = 1;
    Wall         = 2;
    BottleOne    = 3;
    BottleTwo    = 4;
  }

  /** @brief An enumeration for the primary gas used by the machine. */
  enum PrimaryGas
  {
    NullGas  = 0;/**<< Signals not provided, or no change */
    NoGas    = 1;
    Air      = 2;
    Nitrogen = 3;
  }

  /** @brief An enumeration for the primary gas used by the machine. */
  enum Connection
  {
    NullConnection = 0;/**<< Signals not provided, or no change */
    Off            = 1;
    Mask           = 2;
    Tube           = 3;
  }
}